### PR TITLE
fix adding pytest import

### PR DESCRIPTION
**Pull request by Borda**
fixing https://github.com/tmarktaylor/phmdoctest/issues/5#issuecomment-840379311

it turned out that if the check for existing markers was not robust enough :rabbit:

### CODE REVIEW

File: src/phmdoctest/cases.py [hunk 1]
```diff
@@ -87,10 +87,9 @@ def needs_sys(blocks: List[FencedBlock]) -> bool:
 def has_pytest_mark_decorator(blocks: List[FencedBlock]) -> bool:
     """True if any code blocks generate @pytest.mark.* decorators."""
     for block in blocks:
-        if block.role == Role.CODE:
-            for directive in block.directives:
-                if directive.type in [Marker.SKIP, Marker.SKIPIF]:
-                    return True
+        any_mark = any(dr.type in (Marker.MARK_SKIP, Marker.MARK_SKIPIF) for dr in block.directives)
+        if block.role == Role.CODE and any_mark:
+            return True
```
Comment (Borda) [inline src/phmdoctest/cases.py:92]: this is the fix core

File: src/phmdoctest/direct.py [hunk 2]
```diff
@@ -9,8 +9,9 @@
 class Marker(Enum):
     """HTML comment before a fenced code block."""
     SKIP = '<!--phmdoctest-skip-->'
+    SKIPIF = '<!--phmdoctest-skipif<3.'   # No trailing space, no "-->".
     MARK_SKIP = '<!--phmdoctest-mark.skip-->'
-    SKIPIF = '<!--phmdoctest-mark.skipif<3.'   # No trailing space, no "-->".
+    MARK_SKIPIF = '<!--phmdoctest-mark.skipif<3.'   # No trailing space, no "-->".
```
Comment (Borda) [inline src/phmdoctest/direct.py:14]: I assume this was typo in naming

**Comment by tmarktaylor** [comment]
Please let me first apologise for not posting a policy for contributors
and for not having experience with pull requests.
Is it practical to direct this pull request to the new branch v1.1?
If not, I will likely do the following:
- finish reviewing this pull request
- think about a test case to catch the original problem in v1.1
- accept and merge the pull request into master
- merge master into branch v1.1
- update documentation in v1.1
- merge v1.1 back into master
- follow checklist.txt steps to release to PYPI as Version 1.1.1

I prefer master to be as close to latest PYPI release as possible.

**Comment by Borda** [comment]
hmm I see now you branching and feel a bit confused... what I was doing in other projects and in general practice is having 
- master as default and then all releases from this branch with tags (for simple projects)
- master as default and if needed cherry-pick bug-fix commits to a particular release branch - `v1.1.x`
- hold master as stable so you place all release and feature development in `develop` (in fact for small projects the same as the fist one)

**Comment by tmarktaylor** [comment]
The 3rd option "Hold master as stable" meets the need to have the repository README.md be like the latest PYPI release so that when someone visits the repos they see the README for the latest release. Please refer to phmdoctest/doc/about.md.

Would I add a branch called develop then to do this?  Can you show me an example of a project that has develop?

**Comment by Borda** [comment]
> Would I add a branch called develop then to do this? Can you show me an example of a project that has develop?

you can check for example https://github.com/simple-icons/simple-icons

File: src/phmdoctest/direct.py (hunk 2 above)
Comment (tmarktaylor) [reply src/phmdoctest/direct.py:14]: There should only be one enumeration intended to generate @pytest.mark.skipif in the outfile.
I see two here: SKIPIF and MARK_SKIPIF.

File: src/phmdoctest/cases.py (hunk 1 above)
Comment (tmarktaylor) [reply src/phmdoctest/cases.py:92]: - Looks like Marker.SKIP_IF renamed to Marker.MARK_SKIPIF.
- Looks like no other semantic/logic changes.

**Comment by tmarktaylor** [comment]
I entered 3 separate comments under Files Changed | Review Changes.
They are visible on to me on the files changed tab of pytest import #6. 
I am not confident that the review was submitted properly.
There are 2 separate issues that need fixing.
They start with:
- Can't find a caller
-  There should only be

It would be helpful to know the number and order of directives that exposed the bug.

**Comment by tmarktaylor** [comment]
Speculation about the issue that this pull request fixes.

When running phmdoctest on this markdown file like this:

python3 -m phmdoctest t.md --outfile t.py

~~~
<!--phmdoctest-mark.skip-->
<!--phmdoctest-label test_mark_skip-->
```python
print('testing @pytest.mark.skip().')
```
```
incorrect expected output
```
~~~
This python file is generated:
```py3
"""pytest file built from t.md"""
from phmdoctest.functions import _phm_compare_exact


@pytest.mark.skip()
def test_mark_skip(capsys):
    print('testing @pytest.mark.skip().')

    _phm_expected_str = """\
incorrect expected output
```

Which looks very nice but is missing the `import pytest`.

**Comment by Borda** [comment]
@tmarktaylor not sure why you created the alternative PR and do not work with this one, as repo admin you can:
- make any extra commit or change in this PR
- change the destination/target branch as you wish

> Which looks very nice but is missing the `import pytest`.

yes, that is exactly what is this PR about

**Comment by tmarktaylor** [comment]
Cool. Thanks for the instructions:
>  make any extra commit or change in this PR 
>  change the destination/target branch as you wish

**Pull request merged**